gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/AuditReporterManagerImpl.java

```diff
@@ -231,36 +238,38 @@ private void reloadReporter(String reporterId) {
         reporterService.findById(reporterId)
                 .subscribe(
                         reporter -> {
-                            logger.debug("Reload reporter: {} after configuration update", reporter.getName());
-                            Optional<Reporter> optionalAuditReporter = auditReporters
-                                    .entrySet()
-                                    .stream()
-                                    .filter(entry -> reporter.getId().equals(entry.getKey().getId()))
-                                    .map(entry -> entry.getValue())
-                                    .findFirst();
-
-                            if (optionalAuditReporter.isPresent()) {
-                                try {
-                                    Reporter auditReporter = optionalAuditReporter.get();
-                                    // reload the provider if it's enabled
-                                    if (reporter.isEnabled()) {
-                                        auditReporter.stop();
-                                        auditReporters.entrySet().removeIf(entry -> entry.getKey().getId().equals(reporter.getId()));
-                                        loadReporter(reporter);
-                                    } else {
-                                        logger.info("Reporter: {} has been disabled", reporter.getName());
-                                        // unregister event bus consumer
-                                        // we do not stop the underlying reporter if it manages search because it can be used to fetch reportable
-                                        ((EventBusReporterWrapper) auditReporter).unregister();
-                                        if (!auditReporter.canSearch()) {
+                            if (needDeployment(reporter)) {
```

Comment:
Better use of `Optional` (i.e. process the optional in `ifPresentOrElse`, rather than verbosely use an if/else on `Optional.get`):

```suggestion
                           auditReporters
                                    .entrySet()
                                    .stream()
                                    .filter(entry -> reporter.getId().equals(entry.getKey().getId()))
                                    .map(Entry::getValue)
                                    .findFirst()
                                    .ifPresentOrElse(auditReporter -> {
                                        try {
                                            // reload the provider if it's enabled
                                            if (reporter.isEnabled()) {
                                                auditReporter.stop();
                                                auditReporters.entrySet().removeIf(entry -> entry.getKey().getId().equals(reporter.getId()));
                                                loadReporter(reporter);
                                            } else {
                                                logger.info("Reporter: {} has been disabled", reporter.getName());
                                                // unregister event bus consumer
                                                // we do not stop the underlying reporter if it manages search because it can be used to fetch reportable
                                                ((EventBusReporterWrapper) auditReporter).unregister();
                                                if (!auditReporter.canSearch()) {
                                                    auditReporter.stop();
                                                }
                                            }
                                        } catch (Exception e) {
                                            logger.error("An error occurs while reloading reporter: {}", reporter.getName(), e);
                                        }
                                    }, () -> logger.info("There is no reporter to reload"));
```

Reply:
Without the ugly github formatting:

![Screenshot 2022-12-01 at 11 18 57](https://user-images.githubusercontent.com/99647292/205039789-574c29f8-2c9f-4bc4-8629-c4584ef1157b.png)